### PR TITLE
Encryption-at-rest pattern: EBS and EFS

### PR DESCRIPTION
**Pull request by aliaksei-ivanou**
*Issue #, if available:*

*Description of changes:*
Adding a new security blueprint for data-at-rest encryption.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/patterns/security/encryption-at-rest.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# EKS Encryption-at-Rest pattern
+
+## Objective
+
+The objective of this pattern is to demonstrate how to enable encryption-at-rest for EKS cluster using EBS/EFS storage.
```
Comment (shapirov103) [inline docs/patterns/security/encryption-at-rest.md:5]: why is it hyphenated?

File: docs/patterns/security/encryption-at-rest.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+To achieve this objective, the pattern utilizes [EBS CSI Driver Amazon EKS Add-on](https://aws-quickstart.github.io/cdk-eks-blueprints/addons/ebs-csi-driver/) to enable encryption-at-rest for EBS volumes. The pattern also leverages [EFS CSI Driver Amazon EKS Add-on](https://aws-quickstart.github.io/cdk-eks-blueprints/addons/efs-csi-driver/) to enable encryption-at-rest for EFS volumes.
+
+The pattern also leverages KMS resource provider to create a KMS keys for EBS/EFS encryption-at-rest and EFS File System resource provider to create an encrypted EFS file system.
```
Comment (shapirov103) [inline docs/patterns/security/encryption-at-rest.md:9]: minor: "a KMS keys"  - either "a KMS key" or "KMS keys"

File: docs/patterns/security/encryption-at-rest.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## GitOps confguration
+
+For GitOps, the blueprint bootstrap the ArgoCD addon and points to the [EKS Blueprints Workload](https://github.com/aws-samples/eks-blueprints-workloads) sample repository.
```
Comment (shapirov103) [inline docs/patterns/security/encryption-at-rest.md:13]: minor: bootstrap = bootstraps

File: docs/patterns/security/encryption-at-rest.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+The sample repository contains the following workloads:
+
+1. team-platform creates a storage class for EBS and EFS volumes.
+2. team-data creates a persistent volume claim for EBS and EFS volumes and a pod that mounts the volumes.
+
+## Prerequisites
+
+1. Clone the repository
```
Comment (shapirov103) [inline docs/patterns/security/encryption-at-rest.md:22]: minor: consistency in punctuation - each item in the numbered list ends with a "."

File: docs/patterns/security/encryption-at-rest.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+1. Follow the usage [instructions](README.md#usage) to install the dependencies
+1. `argo-admin-password` secret must be defined in Secrets Manager in the same region as the EKS cluster.
```
Comment (shapirov103) [inline docs/patterns/security/encryption-at-rest.md:24]: minor: expected to see numbers 1. 2. 3. but was 1. 1. 1.

File: docs/patterns/security/encryption-at-rest.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+Now you can verify that the EBS and EFS volumes are encrypted by running the following commands:
+
+- EBS
+
+```bash
+❯ kubectl get pvc -n data
+NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS    AGE
+gp2-encrypted-pvc   Bound    pvc-78bd070e-8eba-4b01-a378-462bb806beb3   10Gi       RWO            gp2-encrypted   14m
+
+❯ aws ec2 describe-volumes --region us-east-1 --filters "Name=tag:kubernetes.io/created-for/pv/name,Values=pvc-78bd070e-8eba-4b01-a378-462bb806beb3" --query 'Volumes[*].{VolumeId:VolumeId, Encrypted:Encrypted, KmsKeyId:KmsKeyId}'
```
Comment (shapirov103) [inline docs/patterns/security/encryption-at-rest.md:43]: call out that the customer will need to replace the actual pvc identifier with their own.

**Review by shapirov103 — COMMENTED** [review]
@aliaksei-ivanou  Very nice! I left a few minor comments, i will test it out and once the minor comments are addressed it will be merged.

File: docs/patterns/security/encryption-at-rest.md (hunk 5 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/encryption-at-rest.md:24]: https://www.markdownguide.org/basic-syntax/#ordered-lists

File: docs/patterns/security/encryption-at-rest.md (hunk 1 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/encryption-at-rest.md]: Addressed

File: docs/patterns/security/encryption-at-rest.md (hunk 2 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/encryption-at-rest.md]: Addressed

File: docs/patterns/security/encryption-at-rest.md (hunk 3 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/encryption-at-rest.md]: Addressed

File: docs/patterns/security/encryption-at-rest.md (hunk 4 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/encryption-at-rest.md]: Addressed

File: docs/patterns/security/encryption-at-rest.md (hunk 6 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/encryption-at-rest.md]: Addressed

**Review by shapirov103 — COMMENTED** [review]
@aliaksei-ivanou 1.7.1 is out, please update in package.json, I will retest after it.

**Comment by aliaksei-ivanou** [comment]
> @aliaksei-ivanou 1.7.1 is out, please update in package.json, I will retest after it.

Done

**Review by shapirov103 — APPROVED** [review]
Validated. 

LGTM

**Pull request merged**